### PR TITLE
feat: integrate sentry to server portal

### PR DESCRIPTION
**Pull request by Tzal3x**
**Changes**
- Integrate Sentry to the Logger implementation on the server portal.
- Update logs to send object looking like`{message: String, other-info: string}`.

For staging, I created a walrus-sites-sp-testnet project on Sentry which you can find [here](https://mysten-labs.sentry.io/traces/?environment=development&environment=vercel-preview&project=4508298139271168&statsPeriod=24h).

Note: the logs it contains are also from local development and there are a lot of trash from previous experiments. Need to clear it up later when I get access. 

**Future work**: 
- [x] Enrich logs in the common library. Maybe rephrase the logs too.

### CODE REVIEW

File: portal/server/app/route.ts [hunk 1]
```diff
@@ -5,6 +5,29 @@ import { getDomain, getSubdomainAndPath } from "@lib/domain_parsing";
...
+    Sentry.captureException(new Error(args.message ))
+});
+logger.setWarnPredicate(args => {
+    addLoggingArgsToSentry(args);
+    Sentry.addBreadcrumb({ message: args.message, data: args })
+} );
+logger.setInfoPredicate(args => {
+    addLoggingArgsToSentry(args);
+    Sentry.addBreadcrumb({ message: args.message, data: args })
+} );
```
Comment (Tzal3x) [inline portal/server/app/route.ts:30]: This is the part that the logger predicates are defined for the server portal, so that it utilizes Sentry. 
The `breadcrumbs` are additional information about a trace (i.e. custom logs). 
The `tags` are added to each event so we can query the logs on the Sentry UI.

File: portal/server/next.config.mjs [hunk 2]
```diff
@@ -16,4 +20,37 @@ const nextConfig = {
...
+// https://docs.sentry.io/platforms/javascript/guides/nextjs/manual-setup/
+
+// Upload a larger set of source maps for prettier stack traces (increases build time)
+widenClientFileUpload: true,
+
+// Route browser requests to Sentry through a Next.js rewrite to circumvent ad-blockers.
+// This can increase your server load as well as your hosting bill.
+// Note: Check that the configured route will not match with your Next.js middleware, otherwise reporting of client-
+// side errors will fail.
+tunnelRoute: "/monitoring",
```
Comment (Tzal3x) [inline portal/server/next.config.mjs:43]: Might need to remove the `tunnelRoute`.

File: portal/server/app/route.ts (hunk 1 above)
Comment (giac-mysten) [reply portal/server/app/route.ts:30]: question: isn't the information duplicated if the  tags and the breadcrumbs both contain `args`?  is this useful?

File: portal/server/next.config.mjs (hunk 2 above)
Comment (giac-mysten) [reply portal/server/next.config.mjs:43]: question: why remove?

File: portal/server/instrumentation.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+export async function register() {
+  if (process.env.NEXT_RUNTIME === 'nodejs') {
+    await import('./sentry.server.config');
+  }
+
+  if (process.env.NEXT_RUNTIME === 'edge') {
+    await import('./sentry.edge.config');
+  }
+}
```
Comment (giac-mysten) [inline portal/server/instrumentation.ts:14]: question: do we need both sentry edge and server?

File: portal/server/next.config.mjs (hunk 2 above)
Comment (Tzal3x) [reply portal/server/next.config.mjs:43]: Because it might break the routing of the portal. Users might not be able to define a `/monitoring` route in their walrus site.

Comment (Tzal3x) [reply portal/server/next.config.mjs:43]: Scratch that, I tried it out and there was no conflict with the WS, meaning that the portal will look up for a /monitoring site resource.

File: portal/server/instrumentation.ts (hunk 3 above)
Comment (Tzal3x) [reply portal/server/instrumentation.ts:14]: For local development as far as I understand we need nodejs, while edge for vercel deployments.
